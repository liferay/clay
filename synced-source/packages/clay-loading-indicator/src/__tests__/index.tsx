/**
 * SPDX-FileCopyrightText: Â© 2019 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayLoadingIndicator from '..';
import React from 'react';
import TestRenderer from 'react-test-renderer';

describe('ClayLoadingIndicator', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(<ClayLoadingIndicator />);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the deprecated small variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator small />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the deprecated light variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator light />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the deprecated small and the deprecated light variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator light small />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the circle shape', () => {
		const testRenderer = TestRenderer.create(<ClayLoadingIndicator />);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the squares shape', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator shape="squares" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the extra small variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator size="xs" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the small variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator size="sm" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the medium variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator size="md" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the large variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator size="lg" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the primary variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator displayType="primary" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the secondary variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator displayType="secondary" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders as the light variant', () => {
		const testRenderer = TestRenderer.create(
			<ClayLoadingIndicator displayType="light" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
