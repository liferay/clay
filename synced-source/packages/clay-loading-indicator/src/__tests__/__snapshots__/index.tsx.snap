// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayLoadingIndicator renders 1`] = `
<span
  aria-hidden="true"
  className="loading-animation"
/>
`;

exports[`ClayLoadingIndicator renders as the circle shape 1`] = `
<span
  aria-hidden="true"
  className="loading-animation"
/>
`;

exports[`ClayLoadingIndicator renders as the deprecated light variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-light"
/>
`;

exports[`ClayLoadingIndicator renders as the deprecated small and the deprecated light variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-light loading-animation-sm"
/>
`;

exports[`ClayLoadingIndicator renders as the deprecated small variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-sm"
/>
`;

exports[`ClayLoadingIndicator renders as the extra small variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-xs"
/>
`;

exports[`ClayLoadingIndicator renders as the large variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-lg"
/>
`;

exports[`ClayLoadingIndicator renders as the light variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-light"
/>
`;

exports[`ClayLoadingIndicator renders as the medium variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-md"
/>
`;

exports[`ClayLoadingIndicator renders as the primary variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-primary"
/>
`;

exports[`ClayLoadingIndicator renders as the secondary variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-secondary"
/>
`;

exports[`ClayLoadingIndicator renders as the small variant 1`] = `
<span
  aria-hidden="true"
  className="loading-animation loading-animation-sm"
/>
`;

exports[`ClayLoadingIndicator renders as the squares shape 1`] = `
<span
  aria-hidden="true"
  className="loading-animation-squares"
/>
`;
